bin/http-signature-header-js

```diff
@@ -0,0 +1,162 @@
+      'An HTTP Message must be passed to stdin for this command.');
+  }
+  // this will create a request or response object
+  // similar to node's default request object.
+  const request = httpMessageParser(HTTPMessage);
+  return request;
+}
+
+function makeHTTPHeaders(headers = {}) {
+  let message = '';
+  for(const key in headers) {
+    let value = headers[key];
+    if(Array.isArray(value)) {
+      value = value.join(',');
+    }
+    message += `${key}: ${value}`;
+  }
+  return message;
+}
+
+program
+  .command('c14n')
+  .alias('cannonize')
+  .action(async () => {
+    try {
+      const {headers} = program;
+      const requestOptions = await getHTTPMessage();
+      if(!headers) {throw new Error(
+        '--headers required for command c14n|cannonize' +
+        'ex: --header date,etag');}
+      const includeHeaders = program.headers ? program.headers.split(/\s/) : '';
+      const result = httpSigs.
+        createSignatureString({includeHeaders, requestOptions});
+      console.log(result);
+      process.exit(0);
+    } catch(e) {
+      throw e;
+      process.exit(1);
+    }
+  });
+
+program
+  .command('sign')
+  .action(async cmd => {
+    try {
+      const {
+        headers,
+        keyId,
+        privateKey,
+        keyType,
+        algorithm,
+        created,
+        expires
+      } = program;
+      const privateKeyFile = await readFile(privateKey);
+      const includeHeaders = headers ? program.headers.split(/\s/) : '';
+      const options = {
+        algorithm,
+        includeHeaders,
+        keyId,
+        signature
+      };
+      // TODO: implement a sign function
+      // like this:
+      // https://github.com/digitalbazaar/bedrock-test/blob/master/lib/helpers.js#L22
+      const result = httpSigs.createAuthzHeader(options);
+      throw new Error('command sign has not been implemented yet');
+      console.log(result);
+    } catch(e) {
+      console.error(JSON.stringify(e));
+      process.exit(1);
+    }
+  });
+
+program
+  .command('verify')
+  .option('-s, --signature <signature>', 'Signature.')
+  .action(async ({signature}) => {
+    try {
+      const reqJson = await getHTTPMessage();
+      const {
+        headers,
+        keyId,
+        privateKey,
+        publicKey,
+        keyType,
+        algorithm,
+        created,
+        expires
+      } = program;
+      const publicKeyFile = await readFile(publicKey);
+      // TODO: abstract middleware verify in this.
+      // TODO: get options from program env variables.
+      throw new Error('Not implemented yet.')
+    } catch(e) {
+      console.error('Error:', JSON.stringify(e, null, e), e);
+      process.exit(1);
+    }
+  });
+
+// all of the common options are accessed
+// from program.name i.e. program.headers etc
+program
+  .version('0.0.1')
+  .option('-H, --headers <headers>', 'A list of header names.')
+  .option('-k, --keyId <keyId>', 'A Key Id string.')
+  .option('-private, --private-key <privateKey>.',
```

Comment:
Did listing short flags of `-private`, `-kt`, and `-private` actually work?  I'm pretty sure they have to be single letters since you can combine them like `-a -b -c` === `-abc`.  See https://github.com/tj/commander.js/#option-parsing.

Reply:
looks like you are correct here and good catch!
corrected all of them to single letter uppercase commands.